robotoff/insights/question.py

```diff
@@ -141,16 +148,13 @@ def format_question(self, insight: ProductInsight, lang: str) -> Question:
         )
 
     @staticmethod
-    def get_source_image_url(barcode: str) -> Optional[str]:
-        product: Optional[JSONType] = get_product(barcode, fields=["selected_images"])
+    def get_source_image_url(self, barcode: str) -> Optional[str]:
+        product: Optional[JSONType] = get_product(barcode)
 
         if product is None:
             return None
 
-        if "selected_images" not in product:
-            return None
-
-        selected_images = product["selected_images"]
+        selected_images = self.generate_selected_images(product["images"])
```

Comment:
You can call this method with  `CategoryQuestionFormatter.generate_selected_images`